### ISSUE TITLE
Negative reference count in _gnix_fr_free

### ISSUE THREAD

**Issue by sungeunchoi** [ofi-cray/libfabric-cray#668]
Saw this only once while running `osu_bibw` mid test (after size=1024):

```
% srun -n2 -N2 -t00:2:00 --exclusive ./osu_bibw
# OSU MPI Bi-Directional Bandwidth Test v5.0
# Size      Bandwidth (MB/s)
...
osu_bibw: ./prov/gni/include/gnix_ep.h:171: _gnix_fr_free: Assertion `references_held >= 0' failed.
osu_bibw: ./prov/gni/include/gnix_ep.h:171: _gnix_fr_free: Assertion `references_held >= 0' failed.
```

**Comment by sungeunchoi** [comment]
Seems possibly related to #676

**Comment by sungeunchoi** [comment]
Fixed by #679
